### ISSUE TITLE
Xamarin.Forms iOS app fails to build with airship.ios.core

### ISSUE THREAD

**Issue by alexkozler** [urbanairship/urbanairship-xamarin#219]
```
2>C:\Users\Alex\.nuget\packages\airship.ios.core\16.12.2\build\airship.ios.core.targets(6,9): error MSB3027: Could not copy "C:\Users\Alex\.nuget\packages\airship.ios.core\16.12.2\build\..\content\res\AirshipCore.xcframework\ios-arm64\AirshipCore.framework\Modules\AirshipCore.swiftmodule\arm64-apple-ios.swiftinterface" to "C:\Users\Alex\source\Workspaces\Workspace\InText Xamarin\InTextXam\InText\InText.iOS\bin\iPhoneSimulator\Debug\/AirshipBindings.iOS.Core.resources/AirshipCore.xcframework\ios-arm64\AirshipCore.framework\Modules\AirshipCore.swiftmodule\arm64-apple-ios.swiftinterface". Exceeded retry count of 10. Failed. 

2>C:\Users\Alex\.nuget\packages\airship.ios.core\16.12.2\build\airship.ios.core.targets(6,9): error MSB3021: Unable to copy file "C:\Users\Alex\.nuget\packages\airship.ios.core\16.12.2\build\..\content\res\AirshipCore.xcframework\ios-arm64\AirshipCore.framework\Modules\AirshipCore.swiftmodule\arm64-apple-ios.swiftinterface" to "C:\Users\Alex\source\Workspaces\Workspace\InText Xamarin\InTextXam\InText\InText.iOS\bin\iPhoneSimulator\Debug\/AirshipBindings.iOS.Core.resources/AirshipCore.xcframework\ios-arm64\AirshipCore.framework\Modules\AirshipCore.swiftmodule\arm64-apple-ios.swiftinterface". Could not find a part of the path 'C:\Users\Alex\source\Workspaces\Workspace\InText Xamarin\InTextXam\InText\InText.iOS\bin\iPhoneSimulator\Debug\/AirshipBindings.iOS.Core.resources/AirshipCore.xcframework\ios-arm64\AirshipCore.framework\Modules\AirshipCore.swiftmodule\arm64-apple-ios.swiftinterface'.
```
Really perplexed by this one. I'm trying to update and build my Xamarin.Forms app, with:
- airship.netstandard on the shared NetStandard library
- airship.android.core on the Android app (works fine)
- airship.ios.core on the iOS app, which causes these issues

I also tried uninstalling airship.ios.core and just using airship.ios instead, and it won't even install. I get this:
```
System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\Alex\.nuget\packages\airship.ios.preferencecenter\16.12.2\content\res\AirshipPreferenceCenter.xcframework\ios-arm64_x86_64-simulator\AirshipPreferenceCenter.framework\Modules\AirshipPreferenceCenter.swiftmodule\x86_64-apple-ios-simulator.swiftinterface'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   at System.IO.File.Create(String path)
   at NuGet.Packaging.StreamExtensions.Testable.MmapCopy(Stream inputStream, String fileFullPath, Int64 size)
   at NuGet.Packaging.StreamExtensions.Testable.CopyToFile(Stream inputStream, String fileFullPath)
   at NuGet.Packaging.PackageFileExtractor.ExtractPackageFile(String source, String target, Stream stream)
   at NuGet.Packaging.PackageArchiveReader.CopyFiles(String destination, IEnumerable`1 packageFiles, ExtractPackageFileDelegate extractFile, ILogger logger, CancellationToken token)
   at NuGet.Packaging.PackageReaderBase.CopyFilesAsync(String destination, IEnumerable`1 packageFiles, ExtractPackageFileDelegate extractFile, ILogger logger, CancellationToken cancellationToken)
   at NuGet.Packaging.PackageExtractor.<>c__DisplayClass5_0.<<InstallFromSourceAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.Common.ConcurrencyUtilities.<ExecuteWithFileLockedAsync>d__5`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at NuGet.Common.ConcurrencyUtilities.<ExecuteWithFileLockedAsync>d__5`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.Packaging.PackageExtractor.<InstallFromSourceAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.Commands.ProjectRestoreCommand.<InstallPackageAsync>d__16.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at NuGet.Commands.ProjectRestoreCommand.<InstallPackagesAsync>d__15.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.Commands.ProjectRestoreCommand.<TryRestoreAsync>d__9.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.Commands.RestoreCommand.<ExecuteRestoreAsync>d__85.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.Commands.RestoreCommand.<ExecuteAsync>d__67.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.Commands.RestoreRunner.<ExecuteAsync>d__7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.Commands.RestoreRunner.<CompleteTaskAsync>d__10.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at NuGet.Commands.RestoreRunner.<RunWithoutCommit>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.PackageManagement.DependencyGraphRestoreUtility.<PreviewRestoreProjectsAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.PackageManagement.NuGetPackageManager.<PreviewBuildIntegratedProjectsActionsAsync>d__89.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.PackageManagement.NuGetPackageManager.<PreviewProjectsInstallPackageAsync>d__75.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.PackageManagement.VisualStudio.NuGetProjectManagerService.<>c__DisplayClass22_0.<<GetInstallActionsAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.PackageManagement.VisualStudio.NuGetProjectManagerService.<CatchAndRethrowExceptionAsync>d__33`1.MoveNext()
```

**Comment by alexthekid07** [comment]
Well, false alarm with part of the previous issue... I needed to enable longer path lengths. Problem is now trying to use airship.ios.core I get this instead

```
Failed to resolve assembly: 'AirshipBindings.iOS.Basement, Version=16.12.2.0, Culture=neutral, PublicKeyToken=null'	
```

Xamarin is fun :)

**Comment by jyaganeh** [comment]
Hi @alexthekid07,  it unfortunately looks like we're missing a dependency on basement in the core nuspec. We'll get it fixed up in the next release.

In the meantime, you can work around this by using the combined `airship.ios` package, which should work now that you've enabled long paths. Alternately, if you're only interested in the functionality in `airship.ios.core`, you can add `airship.ios.basement` as an explicit dependency using one of the below options:

Package Ref:
```
<PackageReference Include="airship.ios.basement" Version="16.12.2" />
```

.NET CLI:
```
$ dotnet add package airship.ios.basement --version 16.12.2
```

We can keep this open until we ship a fix, to let you know when you can remove the workaround.

Thanks for the report, and sorry you ran into this!

**Comment by jyaganeh** [comment]
Hey @alexkozler, circling back to let you know that we just shipped v17.1.1 (with iOS 16.12.4 and Android 16.11.1.1 bindings). I'm going to go ahead and close this, since you should be able to remove the workaround after updating. Please feel free to re-open or file a new issue if you run into anything.

**Comment by alexkozler** [comment]
@jyaganeh I believe the same issue is happening with https://github.com/urbanairship/airship-dotnet/issues/5
